tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CustomizedCodeUpdateTool/TspClientUpdateTool.cs

```diff
@@ -0,0 +1,639 @@
+                // Generate a simple diff: replace 'int bar' with 'string bar' in impacted files
+                if (File.Exists(f))
+                {
+                    var content = File.ReadAllText(f);
+                    var newContent = content.Replace("int bar", "string bar");
+                    var diff = $"--- a/{f}\n+++ b/{f}\n@@ -1,1 +1,1 @@\n- {content}\n+ {newContent}\n";
+                    session.ProposedPatches.Add(new PatchProposal
+                    {
+                        File = f,
+                        Diff = diff,
+                        Rationale = "Update for API signature change"
+                    });
+                }
+                else
+                {
+                    session.ProposedPatches.Add(new PatchProposal
+                    {
+                        File = f,
+                        Diff = $"--- a/{f}\n+++ b/{f}\n@@ -1,1 +1,1 @@\n- (file missing)\n+ (new file)\n",
+                        Rationale = "File missing, would create new"
+                    });
+                }
+            }
+            session.Status = "PatchesProposed";
+            SaveSessionToDisk(session);
+            return Task.FromResult(new TspClientUpdateResponse { Session = session, Message = $"Proposed {targetFiles.Count} patches" });
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Propose failed: {sessionId}", sessionId);
+            SetFailure();
+            return Task.FromResult(new TspClientUpdateResponse { ResponseError = ex.Message });
+        }
+    }
+
+    private Task<TspClientUpdateResponse> DoApply(string sessionId, bool dryRun, CancellationToken ct)
+    {
+        try
+        {
+            ct.ThrowIfCancellationRequested();
+            var session = RequireSession(sessionId);
+            if (session.ProposedPatches.Count == 0)
+            {
+                return Task.FromResult(new TspClientUpdateResponse { Session = session, Message = "No patches to apply" });
+            }
+            int applied = 0;
+            foreach (var patch in session.ProposedPatches)
+            {
+                if (dryRun)
+                {
+                    continue;
+                }
+                // For demo: just replace 'int bar' with 'string bar' in the file
+                if (File.Exists(patch.File))
+                {
+                    var content = File.ReadAllText(patch.File);
+                    var newContent = content.Replace("int bar", "string bar");
+                    File.WriteAllText(patch.File, newContent);
+                    applied++;
+                }
+            }
+            session.Status = dryRun ? "AppliedDryRun" : "Applied";
+            SaveSessionToDisk(session);
+            return Task.FromResult(new TspClientUpdateResponse { Session = session, Message = dryRun ? "Dry-run apply complete" : $"Applied {applied} patches" });
```

Comment:
You can make this function `async`, and then just return `new TspClientUpdateResponse { Session = session, Message = dryRun ? "Dry-run apply complete" : $"Applied {applied} patches" };`.

I think you've done this across a few functions, so you can simplify and take advantage of the syntactical sugar. :)